homeassistant/components/apcupsd/config_flow.py

```diff
@@ -49,32 +51,22 @@ async def async_step_user(
         self._async_abort_entries_match({CONF_HOST: host, CONF_PORT: port})
 
         # Test the connection to the host and get the current status for serial number.
-        coordinator = APCUPSdCoordinator(self.hass, host, port)
-        await coordinator.async_request_refresh()
-
-        if isinstance(coordinator.last_exception, (UpdateFailed, TimeoutError)):
+        try:
+            async with asyncio.timeout(CONNECTION_TIMEOUT):
+                data = APCUPSdData(await aioapcaccess.request_status(host, port))
+        except (OSError, asyncio.IncompleteReadError, TimeoutError):
             errors = {"base": "cannot_connect"}
             return self.async_show_form(
                 step_id="user", data_schema=_SCHEMA, errors=errors
             )
 
-        if not coordinator.data:
+        if not data:
```

Comment:
In practice, almost never. The only time this would occur would be that the connection went fine but the daemon somehow returned an empty status (`{}`). But in practice it's almost never the case (it would at the very least return the first `APC` line indicating the version and another `END APC` line to indicate the end of the status).

We could probably remove this defensive logic in a follow-up PR to make the code simpler. Let me know if you disagree or would like this to be handled in this PR, of course!

Reply:
Sounds fine!